pkg/controller/backup/backup_control.go

```diff
@@ -86,7 +86,7 @@ func (bc *defaultBackupControl) removeProtectionFinalizer(backup *v1alpha1.Backu
 	ns := backup.GetNamespace()
 	name := backup.GetName()
 
-	if isDeletionCandidate(backup) && v1alpha1.IsBackupClean(backup) {
+	if !backup.Spec.CleanData || (isDeletionCandidate(backup) && v1alpha1.IsBackupClean(backup)) {
```

Comment:
```suggestion
	if backup.Spec.CleanData && isDeletionCandidate(backup) && v1alpha1.IsBackupClean(backup) {
```

Reply:
Is my suggestion correct?

Reply:
addressed in https://github.com/pingcap/tidb-operator/pull/2892/commits/156e00d9240831138b91865b96c2dcea9a600d10